rust/Dockerfile

```diff
@@ -1,19 +1,23 @@
-FROM rust:1.72-slim as BUILDER
+FROM lukemathwalker/cargo-chef:latest-rust-1.72-slim-bookworm as chef
+WORKDIR /build
+
+FROM chef as planner
+COPY . .
 ARG PACKAGE
-WORKDIR /build/
-COPY . ./
-RUN --mount=type=cache,target=./target \
-    --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked \
-    --mount=type=cache,target=/usr/local/rustup,sharing=locked \
-    cargo build -p $PACKAGE --release
+RUN cargo chef prepare --recipe-path recipe.json
 
-RUN --mount=type=cache,target=./target \
-    mv ./target/release/$PACKAGE /usr/local/bin/$PACKAGE
+FROM chef as builder
+COPY --from=planner /build/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+COPY . .
+ARG PACKAGE
+RUN cargo build -p $PACKAGE --release
+RUN mv ./target/release/$PACKAGE /usr/local/bin/$PACKAGE
```

Comment:
This can be simplified too. Previously, we had to copy the binary out of the cache directory into the container. Now that we just have normal layers, you can copy it from `./target` directly in the final layer.